src/main/java/green/citibike/aws/StationsCache.java

```diff
@@ -0,0 +1,92 @@
+package green.citibike.aws;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import green.citibike.CitiService;
+import green.citibike.CitiServiceFactory;
+import green.citibike.json.StationInfo;
+import green.citibike.json.Stations;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.*;
+
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.lang.reflect.Type;
+import java.time.Duration;
+import java.time.Instant;
+
+public class StationsCache {
+    private static final String KEY = "cache/station_information.json";
+    private static final String BUCKET = "green.citibike";
+    private Stations<StationInfo> response;
+    private Instant lastModified;
+    private Gson gson = new Gson();
+    private S3Client s3Client;
+    private CitiService service = new CitiServiceFactory().getService();
+
+    public StationsCache() {
```

Comment:
You need to go over this code again because it doesn't do what I've asked you to do.